ehcache-impl/src/main/java/org/ehcache/impl/persistence/DefaultLocalPersistenceService.java

```diff
@@ -244,4 +261,33 @@ public File getRoot() {
       return safeSpace.directory;
     }
   }
+
+  private void verifyCleanShutdown(final File rootDirectory) {
+    if (rootDirectory.exists()) {
+      try {
+        if (!cleanShutdown.exists() && !unCleanShutdown.exists()) {
+          if (unCleanShutdown.createNewFile()) {
+            LOGGER.debug("unCleanShutdown file is created.");
+          }
+        } else if (cleanShutdown.exists()) {
+          Files.delete(cleanShutdown.toPath());
+          LOGGER.debug("cleanShutdown file is deleted.");
+          if (unCleanShutdown.createNewFile()) {
+            LOGGER.debug("unCleanShutdown file is created.");
+          }
+        } else if (unCleanShutdown.exists()) {
+          LOGGER.debug("Probably unclean shutdown was done.");
+          if (rootDirectory.exists()) {
+            FileUtils.tryRecursiveDelete(rootDirectory);
```

Comment:
Isn't this recursive delete going to trash the lock file? This might break the existing protection against parallel use of the same directory by multiple instances? We should probably only do any of this marker file manipulation *after* we've validated it is safe for us to access (& modify) the directory.

Reply:
I'm also thinking that making the delete decision will break any users of the LocalPersistenceService that are fault tolerant? Maybe we need to think of a way of signaling the unclean shutdown to consumers of the service, or move this code to a different layer?